### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by foo-jin**
This PR contains the following:
1. All functions now propagate their errors upwards.
2. Errors are now printed in `main` instead of `unwrap`ped
3. The Cargo.toml handling code is replaced by crates to improve the related error messages (these errors are the most common: e.g. trying to use the program outside of a rust repository)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,6 +9,9 @@ version = "0.1.1"
 edition = "2021"
 
 [dependencies]
+anyhow = { version = "1.0" }
```
Comment (kstep) [inline Cargo.toml]: Why not short version (`anyhow = "1.0"`) here instead of braced?

File: src/main.rs [hunk 2]
```diff
@@ -1,71 +1,45 @@
...
-    let output = Command::new("cargo")
-        .arg("locate-project")
-        .output()
-        .unwrap();
-    serde_json::from_str::<CargoLocation>(&String::from_utf8(output.stdout).unwrap())
-        .unwrap()
-        .root
-        .into()
-}
+type Result<T> = anyhow::Result<T>;
```
Comment (kstep) [inline src/main.rs]: Maybe just `pub use anyhow::Result` then?

File: src/main.rs [hunk 3]
```diff
@@ -110,9 +84,18 @@ package() {{
...
-    .unwrap();
+    fn exec() -> Result<()> {
+        let manifest_path = locate_cargo_manifest::locate_manifest()
+            .with_context(|| anyhow!("Failed to locate `{MANIFEST_FILENAME}`"))?;
+        let manifest = parse_manifest(&manifest_path)
+            .with_context(|| anyhow!("Failed to parse `{}`", manifest_path.display()))?;
+        let mut pkgbuild = File::create("PKGBUILD").context("Failed to create PKGBUILD file")?;
+        generate_pkgbuild(&manifest, &mut pkgbuild).context("Failed to generate PKGBUILD")?;
+        Ok(())
+    }
```
Comment (kstep) [inline src/main.rs]: Why not just make `fn main() -> Result<T>` and return from it directly?

**Review by kstep — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 3 above)
Comment (foo-jin) [reply src/main.rs]: You are right, this does have the same behavior. Fixed in f30bb6168e8e313f3b820f8d43282fef58438f8d

File: Cargo.toml (hunk 1 above)
Comment (foo-jin) [reply Cargo.toml]: I think this is a holdover from adding and removing the `backtrace` feature to test what that does.

File: src/main.rs (hunk 2 above)
Comment (foo-jin) [reply src/main.rs]: Fixed in de0f9ad1d6491eaacf9ba7c00d7cc46978e762fc

File: Cargo.toml (hunk 1 above)
Comment (foo-jin) [reply Cargo.toml]: Anyway, fixed in 01c7a4650cde0230df514c219733566fa6150807

**Comment by mfdorst** [comment]
Any movement on this?

**Comment by AkechiShiro** [comment]
Any news on this ? I was going to leave an issue about this bug.
I can take this PR and try and finish it, if that's okay ? @kstep @foo-jin

**Pull request merged**